src/main/java/org/openstreetmap/atlas/checks/validation/points/AddressPointMatchCheck.java

```diff
@@ -0,0 +1,166 @@
+     * @return an optional {@link CheckFlag} object that
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final Point point = (Point) object;
+
+        // Get a bounding box around the Point of interest
+        final Rectangle box = point.getLocation().boxAround(boundsSize);
+
+        // Get all Points in the bounding box, remove Points that have null as their
+        // street name or do not have the street name key tag, and get a set of candidate street
+        // names
+        final Set<String> points = Iterables.stream(point.getAtlas().pointsWithin(box))
+                .filter(nearbyPoint -> nearbyPoint.getTag(POINT_STREET_NAME_KEY).isPresent())
+                .map(nearbyPoint -> nearbyPoint.tag(POINT_STREET_NAME_KEY)).collectToSet();
+
+        // Get all Edges intersecting the bounding box, remove Edges that have null as their
+        // street name or do not have the street name key tag, and get a set of candidate street
+        // names
+        final Set<String> edges = Iterables.stream(point.getAtlas().edgesIntersecting(box))
+                .filter(nearbyEdge -> nearbyEdge.getTag(EDGE_STREET_NAME_KEY).isPresent())
+                .map(nearbyEdge -> nearbyEdge.tag(EDGE_STREET_NAME_KEY)).collectToSet();
+
+        // If there are no Points or Edges in the bounding box
+        if (points.isEmpty() && edges.isEmpty())
+        {
+            // Flag Point with instruction indicating that there are are no suggestions
+            return Optional.of(this.createFlag(point,
+                    this.getLocalizedInstruction(2, point.getOsmIdentifier())));
+        }
+        // If there are Points in the bounding box
+        else if (!points.isEmpty())
+        {
+            // Add all interior Point street names to the list of candidate street names
+            return Optional.of(this.createFlag(point,
+                    this.getLocalizedInstruction(0, point.getOsmIdentifier(), points)));
+        }
+        // If there are Edges intersecting or contained by the bounding box
+        else
+        {
+            return Optional.of(this.createFlag(point,
+                    this.getLocalizedInstruction(1, point.getOsmIdentifier(), edges)));
+        }
+    }
+
+    private boolean hasAssociatedStreetRelation(final AtlasObject object)
+    {
+        // Initialize the rolePresent flag to false as we have not yet determined whether it is
+        // Part of the AssociatedStreet relation
+        boolean rolePresent = false;
+        final Point point = (Point) object;
+
+        // Get all relations that the Point is associated with, filter to keep only relations
+        // Where type=associatedStreet
+        final Set<Relation> relations = point.relations().stream().filter(
+                relation -> relation.tag(RelationTypeTag.KEY).equals(ASSOCIATED_STREET_RELATION))
+                .collect(Collectors.toSet());
```

Comment:
Please consider replacing the logic of this method with something like:
```
point.relations().stream().filter(relation -> relation.tag(RelationTypeTag.KEY).equals(ASSOCIATED_STREET_RELATION))
.anyMatch(relation -> relation.members().stream().anyMatch(member -> member.getRole().equals(STREET_RELATION_ROLE) && member.getEntity().getType().equals(ItemType.EDGE)))
```

Such call would return `true` if point is a part of a relation where relation has an `Edge` member with `street` role.

Java's `Stream` has powerful methods like `anyMatch` that could be used to find if any item is matching the condition in a stream of items. Collecting items into sets here is not necessary.